src/bloomberg/datalake/datalakequerydbconsumer/_data_models.py

```diff
@@ -89,6 +90,17 @@ class ColumnMetrics(Base):  # type: ignore
     __table_args__ = {"extend_existing": True, "schema": "raw_metrics"}
 
 
+def _get_datetime_from_field(field: str | int | float | datetime) -> datetime:
+    if isinstance(field, float):
+        return datetime.fromtimestamp(field)
+    elif isinstance(field, int):
+        return datetime.fromtimestamp(float(field))
+    elif isinstance(field, str):
+        return parser.parse(field)
+    else:
+        return field
```

Comment:
We don't have a runtime guarantee on this, and the error might be confusing. Could we please add an explicit check for datetime and raise an error otherwise? (with some appropriate logging, of course)

Reply:
I added a TypeError as a final check - if it gets raised its caught by _kafka.py:_handle_message and logged there